src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -254,6 +256,50 @@ public OperationalResponse checkout(OperationalRequest request) {
 		return resp;
 	}
 
+	public ResponseEntity<?> totalFavesAndCheckouts(int userId) {
+		OperationalResponse confirm = new OperationalResponse();
+
+		List<User> userFavorites = userRepository.findUserWithUserId(userId, Sort.by("id.productId"));
```

Comment:
This query wouldn't fly. Based on the spec, this API is supposed to return total number of checkouts & list of favorited movies. But findUserWithUserId() returns all the movies associated with the user.

So you need to have a new query in the UserRepository. Something that searches for all the movies a user has favoried. And possibly supports pagination (even though we forgot this in the original spec, I think it's pretty important still).

The query [findCheckedOutMovies()](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/blob/master/src/cs/usfca/edu/histfavcheckout/model/UserRepository.java#L29) does something similar. And it also supports pagination. Just replace checkouts with favorites.